modules/commons/src/main/java/org/apache/synapse/commons/json/JsonUtil.java

```diff
@@ -586,6 +611,29 @@ public static OMElement getNewJsonPayload(MessageContext messageContext, InputSt
                 jsonElement = JSON_OBJECT;
                 messageContext.setProperty(ORG_APACHE_SYNAPSE_COMMONS_JSON_IS_JSON_OBJECT, true);
             }
+            if (isValue) {
+                String jsonString = "";
+                try {
+                    jsonString = IOUtils.toString(json, "UTF-8");
+
+                    messageContext.setProperty(ORG_APACHE_SYNAPSE_COMMONS_JSON_IS_JSON_OBJECT, false);
+                    if (jsonString.startsWith("\"") && jsonString.endsWith("\"")|| jsonString.equals("true") || jsonString.equals("false") ||
+                            jsonString.equals("null") || jsonString.matches("-?\\d+(\\.\\d+)?")) {
+                        jsonElement = JSON_VALUE;
+                    } else {
+                        throw new AxisFault(
+                                "#getNewJsonPayload. Could not save JSON payload. Invalid input stream found. Payload" +
+                                        " is not a JSON string.");
+                    }
+                    OMElement element = OMAbstractFactory.getOMFactory().createOMElement(jsonElement, null);
+                    element.addChild(OMAbstractFactory.getOMFactory().createOMText(jsonString));
+                    return element;
+                } catch (IOException e) {
+                    throw new AxisFault(
+                            "#Can not parse stream. MessageID: " +
+                                    messageContext.getMessageID() + ". Error>>> " + e.getLocalizedMessage());
```

Comment:
I think better if we add the throwable to the new AxisFault
throw new AxisFault("#Can not parse stream. MessageID: " +messageContext.getMessageID() + ". Error>>> " + e.getLocalizedMessage() , e);